.github/workflows/release.yml

```diff
@@ -0,0 +1,123 @@
+name: Release
+on:
+  push:
+    tags:
+      - v*
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+
+jobs:
+  create_draft_release:
+    runs-on: ubuntu-18.04
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Create draft release
+        run: |
+          gh release create \
+            --repo ${{ github.repository }} \
+            --title ${{ github.ref_name }} \
+            --notes '' \
+            --draft \
+            ${{ github.ref_name }}
```

Comment:
If we want to do a nightly build, we should change it to create a release (or pre-release) not a draft?

Reply:
I think it is fine to _start_ with a draft, to have somewhere to upload files to, but we need to end up with a published release. My thinking was publishing a release fires off GitHub notifications (emails etc) so it would be ideal if we only do that after we have all the files.

for nightly build, we need to create the release only _once_ (per minor version) given we’d be mostly re-uploading new files to it every day.

Reply:
And yes, definitely _pre-release_ for nightly. It would be nice to do pre-release for RC tags too.

Reply:
I see.

Reply:
> I think it is fine to start with a draft, to have somewhere to upload files to, but we need to end up with a published release. My thinking was publishing a release fires off GitHub notifications (emails etc) so it would be ideal if we only do that after we have all the files.

But before publish a release. We need to put a changelog to the release which's do it manually at the moment right? It would be nice if we can publish a draft with posting a changelog. :D

Reply:
Yeah, I think doing the release notes and actually publishing would be manual steps for maintainers. Release notes are currently a cut-down version of the changelog, which I think is a good thing (especially for _patch_ releases), so I would not try to automate that part after all.